### PR TITLE
Update tree-sitter to 0.21.0

### PR DESCRIPTION
**Pull request by eed3si9n**
I originally sent a PR to bump to tree-sitter 0.22.1, but maybe that's going too ahead.
Here's a PR to bump to tree-sitter 0.21.0.

Ref https://github.com/neovim/neovim/pull/27566

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,11 +1,11 @@
 [package]
 name = "tree-sitter-scala"
 description = "scala grammar for the tree-sitter parsing library"
-version = "0.20.0"
+version = "0.22.0"
```
Comment (ckipp01) [inline Cargo.toml:4]: Should this be `0.21.0`?

File: package.json [hunk 2]
```diff
@@ -3,23 +3,44 @@
...
+  ],
   "author": "Max Brunsfeld",
   "license": "MIT",
   "dependencies": {
-    "nan": "^2.14.1"
+    "node-addon-api": "^7.1.0",
+    "node-gyp-build": "^4.8.0"
+  },
+  "peerDependencies": {
+    "tree-sitter": "^0.21.0"
```
Comment (ckipp01) [inline package.json:26]: Do we want to lock it at `0.21.0` instead of using `^`?

File: Cargo.toml (hunk 1 above)
Comment (eed3si9n) [reply Cargo.toml]: Addressed in Addressed in 4c7275c70e41c7c1fad6f4e52a60dd3941db5bba

**Review by ckipp01 — APPROVED** [review]
LGTM!

**Pull request merged**